armada/tiller.py

```diff
@@ -99,6 +105,13 @@ def update_release(self, chart, dry_run, name, disable_hooks=False,
         else:
             values = Config(raw=values)
 
+        print name
+        for job in jobs:
+            # LOG.info("Deleting [%s] : %s", "job", items.name)
+            print name, job
+            body = client.V1DeleteOptions()
+            self.k8s.api_client.delete_namespaced_job(name, job, body)
```

Comment:
This assume two restrictive things:

a) all things passed will be jobs
b) all things passed will be deleted

The armada config is more flexible, so the backend needs to be flexible.

the tree is
```
   delete:
     job:
       foobar
```

There could be other trees, e.g. create (maybe CI/CD would want to create mock completed or failed jobs).  I don't think you want to assume on the backend you will delete everything.

I also do not see where you are ensuring you are deleting the job from the namespace of the chart the job is listed under.  In this example, 'openstack'